micro-rdk/src/common/ota.rs

```diff
@@ -442,7 +452,7 @@ impl<S: OtaMetadataStorage> OtaService<S> {
                     total_downloaded += data.len();
 
                     if !got_info {
-                        if total_downloaded < SIZEOF_APPDESC {
+                        if total_downloaded < *FIRMWARE_HEADER_SIZE {
```

Comment:
Micro-nit: Above, we set the max frame size to 16k. If, somehow FIRMWARE_HEADER_SIZE was > 16k, then this would always fail. Maybe the max frame size should be max(16k, FIRMWARE_HEADER_SIZE).

Reply:
I think it's safe to assume (as we're only supporting esp32 at the moment) that a *valid* header will always be <16K. 

We could calculate it out (esp_image_header_t + esp_image_segment_header_t + esp_app_desc_t) and set a magic number accordingly, but I think sizeof provides more context on what we're actually doing. I used 1024 for non-esp32 targets arbitrarily since it's greater than the three types above and it won't have access to the `esp_idf_*` types.

Also, I'm reluctant to set something greater than the 16K as `max_frame_size` since its documentation stated it was the max and would need to look deeper to see what the effects would be inside hyper.